foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/actions/TrendAction.java

```diff
@@ -93,14 +93,10 @@ protected String getRequestCacheKey() {
 
     @Override
     public ActionResponse execute(TrendRequest parameter) throws QueryStoreException {
-        parameter.setTable(ElasticsearchUtils.getValidTableName(parameter.getTable()));
         if (null == parameter.getFilters()) {
             parameter.setFilters(Lists.<Filter>newArrayList(new AnyFilter(parameter.getTable())));
         }
         String field = parameter.getField();
-        if (parameter.getTable() == null) {
```

Comment:
Why is this check removed ? I think table can still be null if validate() call is not made before this line.